node-graph/gstd/src/raster.rs

```diff
@@ -174,6 +174,66 @@ fn compute_transformed_bounding_box(transform: DAffine2) -> Bbox {
 	}
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct InsertChannelNode<P, S, Stencil, TargetChannel> {
+	stencil: Stencil,
+	target_channel: TargetChannel,
+	_p: PhantomData<P>,
+	_s: PhantomData<S>,
+}
+
+#[node_macro::node_fn(InsertChannelNode<_P, _S>)]
+fn insert_channel_node<
+	// _P is the color of the input image. It must have an alpha channel because that is going to
+	// be modified by the mask
+	_P: Copy + Alpha + RGB,
+	// _S is the color of the stencil. It must have a luminance channel because that is used to
+	// mask the input image
+	_S: Luminance,
+	// The channel of the input image that is used to insert the stencil
+	// Input image
+	Input: Transform + RasterMut<Pixel = _P>,
+	// Stencil
+	Stencil: Transform + Sample<Pixel = _S>,
+>(
+	mut image: Input,
+	stencil: Stencil,
+	target_channel: RedGreenBlue,
+) -> Input {
+	let image_size = DVec2::new(image.width() as f64, image.height() as f64);
+	let mask_size = stencil.transform().decompose_scale();
+
+	if mask_size == DVec2::ZERO {
+		return image;
+	}
+
+	// Transforms a point from the background image to the forground image
+	let bg_to_fg = image.transform() * DAffine2::from_scale(1. / image_size);
+	let area = bg_to_fg.transform_point2(DVec2::new(1., 1.)) - bg_to_fg.transform_point2(DVec2::ZERO);
```

Comment:
In my understanding of maths this whole line is just the same as:
```rust
let area = bg_to_fg.transform_vector2(DVec2::ONE);
```
(`.transform_vector2` is an alias of `.matrix2 *` and ignores the translation)

Reply:
Done in next Push :+1: